echo-pubsub-google/src/main/java/com/netflix/spinnaker/echo/config/GooglePubsubProperties.java

```diff
@@ -79,6 +83,29 @@ public InputStream readTemplatePath() {
     }
   }
 
+  @Data
+  @NoArgsConstructor
+  public static class GooglePubsubPublisherConfig {
+
+    @NotEmpty
+    private String name;
+
+    @NotEmpty
+    private String project;
+
+    // Optional. Uses Application Default Credentials if not set.
+    private String jsonPath;
+
+    @NotEmpty
+    private String topicName;
```

Comment:
Would be nice if you could control the kinds of messages that were sent to each topic (i.e. default is send all messages, but you could also do things like 'send only pipeline end events').

Reply:
The initial goal of this was to replicate the REST "all events" firehose, just with pubsub, but I agree that would be a nice enhancement.

Reply:
Sounds great!

Reply:
This is part of the reason I filed https://github.com/spinnaker/spinnaker/issues/3194 - to make what you're suggesting a no brainer.